src/main/java/com/mongodb/jdbc/MongoSQLDatabaseMetaData.java

```diff
@@ -129,33 +267,437 @@ public ResultSet getSchemas() throws SQLException {
+        resultSchema.addRequiredScalarKeys(
+                new Pair<>(TABLE_CAT, BSON_STRING_TYPE_NAME),
+                new Pair<>(TABLE_SCHEM, BSON_STRING_TYPE_NAME),
+                new Pair<>(TABLE_NAME, BSON_STRING_TYPE_NAME),
+                new Pair<>(GRANTOR, BSON_STRING_TYPE_NAME),
+                new Pair<>(GRANTEE, BSON_STRING_TYPE_NAME),
+                new Pair<>(PRIVILEGE, BSON_STRING_TYPE_NAME),
+                new Pair<>(IS_GRANTABLE, BSON_STRING_TYPE_NAME));
+
+        // All fields in this result set are nested under the bottom namespace.
+        MongoJsonSchema botSchema = MongoJsonSchema.createEmptyObjectSchema();
+        botSchema.properties.put(BOT_NAME, resultSchema);
+
+        // Note: JDBC has Catalogs, Schemas, and Tables: they are three levels of organization.
+        // MongoDB only has Databases (Catalogs) and Collections (Tables), so we ignore the
+        // schemaPattern argument.
+        Pattern tableNamePatternRE = Pattern.compile(tableNamePattern);
+
+        Stream<BsonDocument> docs;
+        if (catalog == null) {
+            // If no catalog (database) is specified, get table privileges for all databases.
+            docs =
+                    this.conn
+                            .mongoClient
+                            .listDatabaseNames()
+                            .into(new ArrayList<>())
+                            .stream()
+                            .sorted()
+                            .flatMap(
+                                    dbName ->
+                                            getTablesFromDB(
+                                                    dbName,
+                                                    tableNamePatternRE,
+                                                    null,
+                                                    this::toGetTablePrivilegesDoc));
+        } else {
+            docs =
+                    getTablesFromDB(
+                            catalog, tableNamePatternRE, null, this::toGetTablePrivilegesDoc);
+        }
+
+        // Per JDBC spec, sort by  TABLE_CAT, TABLE_SCHEM (omitted), TABLE_NAME, and
+        // PRIVILEGE. Since the stream is already sorted by TABLE_CAT at this point
+        // and all PRIVILEGEs are the same, we just sort by TABLE_NAME here.
+        List<BsonDocument> docsList =
+                docs.sorted(
+                                Comparator.comparing(
+                                        (BsonDocument doc) ->
+                                                doc.getDocument(BOT_NAME).getString(TABLE_NAME)))
+                        .collect(Collectors.toList());
+
+        BsonExplicitCursor c = new BsonExplicitCursor(docsList);
+
+        return new MongoSQLResultSet(null, c, botSchema);
     }
 
     @Override
     public ResultSet getBestRowIdentifier(
             String catalog, String schema, String table, int scope, boolean nullable)
             throws SQLException {
+        // TODO
```

Comment:
like I said in the PR summary -- working on these remaining 3 still. I cut out the current progress for this milestone so it didn't look too cluttered and confusing.